### PR TITLE
feat:Splitting website data

### PR DESCRIPTION
**Pull request by wangxinbiao**
#### Which issue(s) this PR fixes

Feat:#641 
Feat:#595

### CODE REVIEW

File: pypi/data-processing/src/utils/async_url_playwright_util.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    except Exception:
+        logger.error(
+            ''.join(
+                [
+                    f"{log_tag_const.WEB_CRAWLING} Execute crawling url failure\n",
+                    f"The tracing error is: \n{traceback.format_exc()}"
+                ]
+            )
+        )
+        return {"status": 400, "message": "获取网页中的子网页url失败", "data": ""}
```
Comment (nkwangleiGIT) [inline pypi/data-processing/src/utils/async_url_playwright_util.py:145]: for internal error, the status code should be 500, not 400.

File: pypi/data-processing/src/utils/async_url_playwright_util.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+            # 在浏览器中打开网页
+            await page.goto(url)
+
+            # 提取每个 a 标签的 href 属性
+            links = await page.query_selector_all('a')
+            for link in links:
+                href = await link.get_attribute('href')
+                # 需要抓取的url数量不得超过最大数量
+                if url_count >= max_count:
```
Comment (nkwangleiGIT) [inline pypi/data-processing/src/utils/async_url_playwright_util.py:120]: add some log here to indicate it's beyond the max count

File: pypi/data-processing/src/utils/async_url_playwright_util.py (hunk 1 above)
Comment (wangxinbiao) [reply pypi/data-processing/src/utils/async_url_playwright_util.py]: Modified

File: pypi/data-processing/src/utils/async_url_playwright_util.py (hunk 2 above)
Comment (wangxinbiao) [reply pypi/data-processing/src/utils/async_url_playwright_util.py:127]: already added

File: pypi/data-processing/src/data_store_process/minio_store_process.py [hunk 3]
```diff
@@ -170,6 +170,18 @@ def text_manipulate(
                     task_id=id,
                     create_user=req_json["creator"],
                 )
+            elif file_extension == "web":
+                # 处理.web文件
+                result = await web_handle.text_manipulate(
```
Comment (bjwswang) [inline pypi/data-processing/src/data_store_process/minio_store_process.py:175]: @wangxinbiao as discussed,the `.web` file have the `web crawler configurationas`. Take this as as an example: 


```json
{"url":"http://1.1.1.1","interval_time":500,"resource_types":["type1","type2"],"max_depth":5,"max_count":10,"exclude_sub_urls":["url1","url2"],"include_sub_urls":["url3","url4"],"exclude_img_info":{"weight":100,"height":200}}
```

Comment (wangxinbiao) [reply pypi/data-processing/src/data_store_process/minio_store_process.py:175]: ok

File: pypi/data-processing/src/file_handle/web_handle.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    file_name,
+    document_id,
+    support_type,
+    conn_pool,
+    task_id,
+    create_user,
+    chunk_size=None,
+    chunk_overlap=None,
+):
+    """Manipulate the text content from a pdf file.
```
Comment (bjwswang) [inline pypi/data-processing/src/file_handle/web_handle.py:42]: from a web file

File: pypi/data-processing/src/file_handle/web_handle.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    chunk_overlap: chunk overlap;
+    """
+    logger.debug(f"{log_tag_const.PDF_HANDLE} Start to manipulate the text in web")
+
+    try:
+        pdf_file_path = file_utils.get_temp_file_path()
+        file_path = pdf_file_path + "original/" + file_name
+
+        # Text splitter
+        documents = await _get_documents_by_langchain(
```
Comment (bjwswang) [inline pypi/data-processing/src/file_handle/web_handle.py:58]: 这个地方实际做了两件事情:
1.处理.web文件，封装web爬取参数
2. 爬取网页，并分割
建议将此方法拆分一下：
1. crawler = buildWebCrawler()
2. htmlfiles  = cralwer.run()
3.  post process html

File: pypi/data-processing/src/file_handle/web_handle.py [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+import ulid
+from langchain.text_splitter import SpacyTextSplitter
+from langchain_community.document_transformers import Html2TextTransformer
+
+from common import log_tag_const
+from common.config import config
+from database_operate import data_process_document_chunk_db_operate
+from document_loaders.async_chromium import AsyncChromiumLoader
+from file_handle import common_handle
+from utils import async_url_playwright_util, file_utils, json_utils
```
Comment (bjwswang) [inline pypi/data-processing/src/file_handle/web_handle.py:28]: 可以封装一个WebCrawler extends Langchain's  Playwright Loader

**Comment by wangxinbiao** [comment]
@bjwswang 封装了一个AsyncPlaywrightLoader进行web数据爬取。

File: pypi/data-processing/src/file_handle/common_handle.py [hunk 7]
```diff
@@ -126,11 +126,18 @@ def text_manipulate(
```
Comment (bjwswang) [inline pypi/data-processing/src/file_handle/common_handle.py:126]: It's better to define a common `BaseTextManipulator`

**Review by bjwswang — APPROVED** [review]

File: pypi/data-processing/src/file_handle/common_handle.py (hunk 7 above)
Comment (bjwswang) [reply pypi/data-processing/src/file_handle/common_handle.py:126]: we can do this later.

**Pull request merged**